app/mixins/repo-sync-progress.js

```diff
@@ -0,0 +1,106 @@
+import { alias } from '@ember/object/computed';
+import Mixin from '@ember/object/mixin';
+import { computed, get } from '@ember/object';
+
+const TOTAL_STEPS = 9;
+
+export default Mixin.create({
+  githubRepo: null,
+  projectGithubRepo: null,
+  triggeredSync: false,
+
+  errored: computed('projectRepoState', 'repoState', function() {
+    let projectRepoState = get(this, 'projectRepoState');
+    let repoState = get(this, 'repoState');
+    return (repoState && repoState.indexOf('error') > -1)
+        || (projectRepoState && projectRepoState.indexOf('error') > -1);
+  }),
+
+  percentage: computed('stepNumber', function() {
+    let stepNumber = get(this, 'stepNumber');
+    return (stepNumber / TOTAL_STEPS) * 100;
+  }),
+
+  projectRepoState: alias('projectGithubRepo.syncState'),
+  repoState: alias('githubRepo.syncState'),
+
+  stepNumber: computed('projectRepoState', 'repoState', 'triggeredSync', function() {
+    let projectRepoState = get(this, 'projectRepoState');
+    let repoState = get(this, 'repoState');
+
+    let state = projectRepoState;
+
+    if (!state
+        || state === 'unsynced'
+        || state === 'syncing_github_repo'
+        || state === 'errored_syncing_github_repo') {
+      state = repoState;
+    }
+
+    if (!projectRepoState && repoState === 'receiving_webhooks' && get(this,  'triggeredSync')) {
+      state = 'unsynced';
+    }
+
+    switch (state) {
+      case 'unsynced':
+        return 0;
+      case 'fetching_pull_requests':
+      case 'errored_fetching_pull_requests':
+        return 1;
+      case 'syncing_github_pull_requests':
+      case 'errored_syncing_github_pull_requests':
+        return 2;
+      case 'fetching_issues':
+      case 'errored_fetching_issues':
+        return 3;
+      case 'syncing_github_issues':
+      case 'errored_syncing_github_issues':
+        return 4;
+      case 'fetching_comments':
+      case 'errored_fetching_comments':
+        return 5;
+      case 'syncing_github_comments':
+      case 'errored_syncing_github_comments':
+      case 'receiving_webhooks':
+        return 6;
+      case 'syncing_tasks':
+      case 'errored_syncing_tasks':
+        return 7;
+      case 'syncing_comments':
+      case 'errored_syncing_comments':
+        return 8;
+      case 'synced':
+        return 9;
+      default:
+        return 0;
+    }
+  }),
+
+  shouldShowSync: computed('repoState', 'projectRepoState', 'triggeredSync', function() {
```

Comment:
Something irks me about names like these. It feels sort of like naming a boolean variable `isTrue`.

Yes, the variable tells us we should show the sync UI, but it doesn't tell us why. How about `syncIncomplete` or `syncRequired`. I don't think any of these are perfect, but there's just something off to me with `shouldShowSync`.